src/rosdiscover/models/plugins/controller_manager.py

```diff
@@ -0,0 +1,158 @@
+        context: NodeContext
+            provides access to the node context for the associated controller_manager node
+            that hosts this controller
+        """
+        ...
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class JointStateControllerPlugin(ControllerManagerPlugin):
+    def _load(self, interpreter: Interpreter, context: NodeContext) -> None:
+        root_ns = namespace(self.namespace)
+        context.read("joints", None)
+        context.read("publish_rate")
+        context.read("extra_joints")
+        context.pub(f"{root_ns}/joint_states", "sensor_msgs::JointState")
+
+
+@attr.s(auto_attribs=True, frozen=True, slots=True)
+class DiffDriveControllerPlugin(ControllerManagerPlugin):
+    def _load(self, interpreter: Interpreter, context: NodeContext) -> None:
+        # FIXME this is only a partial model!
+        # https://github.com/ros-controls/ros_controllers/blob/noetic-devel/diff_drive_controller/src/diff_drive_controller.cpp
+
+        # TODO get wheel names
+
+        context.read("publish_rate", 50.0)
+        context.read("open_loop", False)
+        context.read("wheel_separation_multiplier", 1.0)
+
+        if context.hasParam("wheel_radius_multiplier"):
+            context.read("wheel_radius_multiplier")
+        else:
+            context.read("left_wheel_radius_multiplier", 1.0)
+            context.read("right_wheel_radius_multiplier", 1.0)
+
+        context.read("velocity_rolling_window_size", 10)
+        context.read("cmd_vel_timeout", 0.5)
+        context.read("allow_multiple_cmd_vel_publishers", True)
+
+        context.read("base_frame_id", "base_link")
+        context.read("odom_frame_id", "odom")
+        context.read("enable_odom_tf", True)
+
+        context.read("linear/x/has_velocity_limits", False)
+        context.read("linear/x/has_acceleration_limits", False)
+        context.read("linear/x/has_jerk_limits", False)
+        context.read("linear/x/max_velocity", 0.0)
+        context.read("linear/x/min_velocity", 0.0)
+        context.read("linear/x/max_acceleration", 0.0)
+        context.read("linear/x/min_acceleration", 0.0)
+        context.read("linear/x/max_jerk", 0.0)
+        context.read("linear/x/min_jerk", 0.0)
+
+        context.read("angular/z/has_velocity_limits", False)
+        context.read("angular/z/has_acceleration_limits", False)
+        context.read("angular/z/has_jerk_limits", False)
+        context.read("angular/z/max_velocity", 0.0)
+        context.read("angular/z/min_velocity", 0.0)
+        context.read("angular/z/max_acceleration", 0.0)
+        context.read("angular/z/min_acceleration", 0.0)
+        context.read("angular/z/max_jerk", 0.0)
+        context.read("angular/z/min_jerk", 0.0)
+
+        should_publish_command = context.read("publish_cmd", False)
+        should_publish_joint_state = context.read("publish_wheel_joint_controller_state", False)
+
+        context.read("wheel_separation", 0.0)
```

Comment:
This doesn't seem to publish/subscribe to the topics mentioned in http://wiki.ros.org/diff_drive_controller